api/applib/src/main/java/org/apache/causeway/applib/services/repository/RepositoryService.java

```diff
@@ -84,6 +84,14 @@ public interface RepositoryService {
      */
     <T> T detachedEntity(@NonNull T entity);
 
+    /**
+     * Suspends flushing transaction for {@param aClass} instances until the {@param bulkMode} is turned off again.
+     * <p>
+     * Usage should be wrapped in a try {} finally {} construction.
+     *
+     */
+    <T extends Class> void setBulkMode(final T aClass, final Boolean bulkMode);
```

Comment:
Don't like this design, it runs the risk of leaving the service in the 'bulk' mode.  Also, while the impl is on a thread-local, the API suggests that the service is stateful, which is a no-no.

Instead, provide an API that accepts a lambda and does all the work within, eg:

`execInBulk( Callable )`

Reply:
Fixed